[Yuan Jing] iP

seewhyjay opened the pull request:
# Duke

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,

*  text-based
*  easy to learn
*  ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/seewhyjay/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:

```java
    private void run() {
        ui.showWelcome();
        String cmd = ui.readCmd();
        while (parser.parseCommand(cmd)) {
            cmd = ui.readCmd();
            storage.saveTasks(tasks);
        }
        ui.closeScanner();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

seewhyjay committed:
Level-0

seewhyjay committed:
Level-1

seewhyjay committed:
Level-2

seewhyjay committed:
Level-3

seewhyjay committed:
Level-4

seewhyjay committed:
Level-4

seewhyjay committed:
Level-5

seewhyjay committed:
Level-5

seewhyjay committed:
Level-6

seewhyjay committed:
A-Enums

seewhyjay committed:
Fix A-TextUITesting

seewhyjay committed:
Fix Level-4, tasks with spaces work now

seewhyjay committed:
Feat: Implement saving and loading
Feat: Update TextUITest to include negative tests
Chore: Make printing of line separators more consistent

seewhyjay committed:
Merge branch 'branch-Level-7'

seewhyjay committed:
Feat: Implement Dates and Times
Chore: Updated test cases
Fix: Catch when descriptions or dates are empty

seewhyjay committed:
Chore: Remove redundant variables

seewhyjay committed:
Merge branch 'branch-Level-8'

seewhyjay committed:
Feat: A-MoreOOP
Split monolithic Duke class into Ui, Parser, Storage and TaskList classes.
Added DukeExceptions.

seewhyjay committed:
Fix: Save tasks after every command

seewhyjay committed:
Merge branch 'branch-A-MoreOOP'

seewhyjay committed:
Chore: Move all files to Duke package

seewhyjay committed:
Merge branch 'branch-A-Packages'

seewhyjay committed:
Merge branch 'add-gradle-support'

seewhyjay committed:
Chore: Change main class in build.gradle

seewhyjay committed:
Fix: Throw DukeInvalidDateException when using wrong datetime format instead of defaulting to current datetime.
Chore: Rename completed to isCompleted in Task

seewhyjay committed:
Fix: Update runtest.bat to build from the duke package

seewhyjay committed:
Chore: Move printing task list to Ui

seewhyjay committed:
Feat: Implement JUnit tests for Event and TaskList classes

seewhyjay committed:
Chore: JavaDocs

seewhyjay committed:
Feat: Add `find` command to find tasks with keyword
Chore: Update `task` in `Task` to `description`.
Fix: Fixed JUnit test for deletion to check for `IndexOutOfBounds`

seewhyjay committed:
Merge branch 'branch-A-JavaDoc'

seewhyjay committed:
Chore: Rename test functions

seewhyjay committed:
Merge branch 'branch-A-CodingStandard'

seewhyjay committed:
Merge branch 'branch-Level-9'
Chore: Resolve merge conflicts
Chore: Add JavaDoc for `find`

jingjie88 commented on src/test/java/duke/TestTaskList.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    @Test
+    public void taskList_testMarkAndUnmarkTask() {
```
Perhaps it would be better to follow the naming convention given in the notes unitBeingTested_descriptionOfTestInputs_expectedOutcome

jingjie88 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,183 @@
+/**
+ * Represents the user interface for Duke.
+ */
+public class Ui {
```
I like the consistent and clear naming of the methods here 👍

jingjie88 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,104 @@
+                while ((line = reader.readLine()) != null) {
+                    String[] datum = line.split("\\|");
+                    try {
+                        switch (datum[0]) {
```
I think the case statements are supposed to be on the same indentation level as switch according to the coding standard.

jingjie88 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,61 @@
+/**
+ * Represents an event task with a description, start time, and deadline. It inherits from the Task class.
+ */
+public class Event extends Task{
```
I like how clean the code is here 👍

jingjie88 reviewed (commented):
LGTM! Just a few minor coding standard violations.

seewhyjay committed:
Feat: Add GUI for Duke

seewhyjay committed:
Chore: Fix lint issues

seewhyjay committed:
Merge branch 'branch-Level-10'

seewhyjay committed:
Chore: Fix switch case indentation levels

seewhyjay committed:
Feat: Add assert statements.

In this commit, I have added assert statements to enhance the reliability of the codebase. These assertions help in catching potential issues early during development and provide a form of self-documentation for the expected behavior of the code.

The added assertions cover critical parts of the code, ensuring that conditions and assumptions are met as the program executes. This proactive approach to error detection can save debugging time in the long run and make the code more robust.

seewhyjay committed:
Chore: Remove dead code

seewhyjay committed:
Merge pull request #1 from seewhyjay/branch-A-Assertations

Feat: Add assert statements.

seewhyjay committed:
Merge branch 'master' into branch-A-CodeQuality

seewhyjay committed:
Merge pull request #2 from seewhyjay/branch-A-CodeQuality

Chore: Remove dead code

seewhyjay committed:
Feat: Implement `duplicate` command that deletes duplicates

Implemented hashCode for all children of `Task` in order to use hashSet to check for duplicates.

seewhyjay committed:
Chore: Add JavaDocs for hashCode methods

seewhyjay committed:
Merge branch 'branch-C-DetectDuplicates'

seewhyjay committed:
Chore: Update User Guide

seewhyjay committed:
Chore: Fix formatting

seewhyjay committed:
Chore: Fix formatting

seewhyjay committed:
Chore: Update docs for missing feature

seewhyjay committed:
Fix: Tasks save now after every command that changes task list.

seewhyjay committed:
Chore: FIx UI class

seewhyjay committed:
Feat: Refactor TaskList and Parser

seewhyjay committed:
Chore: Update build.gradle

Followed https://github.com/nus-cs2103-AY2324S1/forum/issues/173

seewhyjay committed:
Merge branch 'feature/refactor-parser'

seewhyjay committed:
Chore: Remove unused imports

seewhyjay committed:
Fix: Force gradle to build for all architectures

The previous fix wasn't working for M1 Macs

seewhyjay committed:
Chore: Update README to add ToC and Quick Start

seewhyjay committed:
Chore: No double click to open

seewhyjay committed:
Chore: Update docs
